src/Database/Driver/Mysql.php

```diff
@@ -245,6 +272,28 @@ public function getVersion(): string
         return $this->_version;
     }
 
+    /**
+     * Returns true if the server supports common table expressions.
+     *
+     * @return bool
+     */
+    public function supportsCommonTableExpressions(): bool
+    {
+        if ($this->supportsCommonTableExpressions === null) {
+            $version = $this->getVersion();
+            if (strpos($version, 'MariaDB') !== false) {
```

Comment:
We should make this a `isMariaDb()` helper if we're going to support it.

Reply:
We could use it for window functions if we keep this.

Reply:
Would you prefer if the version string was prepared accordingly then, so that it returns the MariaDB version number only? Currently for MariaDB it can return something like this:

```
5.5.5-10.3.22-MariaDB-0ubuntu0.16.04.1
```

where `5.5.5` is the MySQL version that is being replicated, and `0ubuntu0.16.04.1` either the build environment or the server environment, not really sure.

Reply:
Perhaps an additional method for specifically retrieving the MariaDB version might make sense, something like `getMariaDbVersion()`, which, given the above version string, would return `10.3.22`, while `getVersion()` would return `5.5.5`?

Reply:
I don't know mariadb, but I think returning the maria-db specific version is best. Relying on (probably) outdated emulation version would be trouble.

`isMariaDb()` and `version()`

Reply:
I'm not overly familiar with MariaDB either, but that sounds like a fair point.

Reply:
I could squeeze this in here, but maybe it should rather be in a separate PR, and accompanied by adding at least _one_ test run with an actual MariaDB backend?

Reply:
I think we'd leave out MariaDB-specific code and expect that users would extend the Mysql driver to provide specific checks/support.